libs/src/services/ethContracts/index.js

```diff
@@ -322,19 +329,21 @@ class EthContracts {
    */
   async validateDiscoveryProviderHealth (endpoint) {
     try {
+      const startTime = Date.now()
```

Comment:
fyi---we do already have some timing logic for creator node timing https://github.com/AudiusProject/audius-protocol/blob/master/libs/src/utils.js#L137

also, do we run the risk of not load balancing our discprovs if we roll this out? I think it should be like round robin for discprovs that respond in under X ms

Reply:
@raymondjacobson that's the problem with this approach. it's absolute with no load balancing. what do you think is a reasonable response time. 1 second?

Reply:
yeah 1s sounds reasonable, maybe we do best < 1s and then pick fastest after

Reply:
+1 on the round robin / threshold idea, this can be trivially done by aggressively lowering that axios request timeout